src/guides/lens-protocol/index.md

```diff
@@ -0,0 +1,638 @@
+    const lensHub = await LensHub.at(addrs['lensHub proxy']);
+  
+    await lensHub.setState(ProtocolState.Unpaused, {from: governance});
+    await lensHub.whitelistProfileCreator(user, true, {from: governance});
+    
+    // Will fail if you've already minted this profile
+    // const inputStruct = {
+    //   to: user,
+    //   handle: 'zer0dot',
+    //   imageURI:
+    //     'https://ipfs.fleek.co/ipfs/ghostplantghostplantghostplantghostplantghostplantghostplan',
+    //   followModule: ZERO_ADDRESS,
+    //   followModuleInitData: [],
+    //   followNFTURI:
+    //     'https://ipfs.fleek.co/ipfs/ghostplantghostplantghostplantghostplantghostplantghostplan',
+    // };
+    // await lensHub.createProfile(inputStruct, {from: user});
+  
+    const data = defaultAbiCoder.encode(['uint256'], ['42069']);
+    const secretCodeFollowModule = await SecretCodeFollowModule.deployed();
+    await lensHub.setFollowModule(1, secretCodeFollowModule.address, data, {from: user});
+    
+    try {
+      await lensHub.follow([1], [badData], {from: user, gas: "0xfffff"});
+    } catch (e) {
+      console.log(`Expected failure occurred! Error: ${e}`);
+    }
+    await lensHub.follow([1], [data], {from: user});
+  
+    const followNFTAddr = await lensHub.getFollowNFT(1, {from: governance});
+    const followNFT = await FollowNFT.at(followNFTAddr);
+
+    const totalSupply = await followNFT.totalSupply({from: user});
+    const ownerOf = await followNFT.ownerOf(1, {from: user});
+  
+    console.log(`Follow NFT total supply (should be 1): ${totalSupply}`);
+    console.log(
+      `Follow NFT owner of ID 1: ${ownerOf}, user address (should be the same): ${user}`
+    );
+  } catch(err) {
+    console.log(err);
+  }
+  cb();
+}
+
+module.exports = main;
+```
+
+Replace `[SecretFollowModuleAddress]` in line 36 with the contract address. You can easily find it by running `truffle networks`.
+
+If you haven't executed the `create-profile` script yet, you can uncomment the profile creation piece of this code. Otherwise, if you have created a profile, leave that portion commented, since you cannot create two profiles with the same username.
+
+To run, simply call:
+
+```shell
+truffle exec scripts/secret_follow.js
+```
+
+You should see something a bit like this:
+
+```shell
+Using network 'development'.
+
+Expected failure occurred! Error: StatusError: Transaction: 0x1c22b1e9b35d6531b22e22d807d58be55c96a81e343bf2bb3f5bd35145a1b255 exited with an error (status 0). Reason given: Custom error (could not decode).
+     Please check that the transaction:
+     - satisfies all conditions set by Solidity `require` statements.
+     - does not trigger a Solidity `revert` statement.
+
+Follow NFT total supply (should be 1): 1
+Follow NFT owner of ID 1: 0xA9A3b27098f4446a1019F75e1164F4ca1980727e, user address (should be the same): 0xA9A3b27098f4446a1019F75e1164F4ca1980727e
+```
+
+The first failure is expected because we intentionally input the wrong password!
```

Comment:
Optional...I feel like it ends a little abruptly, might be useful to briefly summarize what's just been and possibly some ideas for additional extensions?